pkg/apis/build/v1alpha1/build_validation.go

```diff
@@ -92,3 +96,43 @@ func (bt *BuildSpec) validateTimeout() *apis.FieldError {
 	}
 	return nil
 }
+
+// Validate source
+func (bs BuildSpec) validateSources() *apis.FieldError {
+	var subPathExists bool
+	var emptyTargetPath bool
+	names := map[string]string{}
+	nodeMap := map[string]*node{}
+
+	// both source and sources cannot be defined in build
+	if len(bs.Sources) > 0 && bs.Source != nil {
+		return apis.ErrMultipleOneOf("b.spec.source", "b.spec.sources")
+	}
+
+	for _, source := range bs.Sources {
+		// check all source have unique names
+		if _, ok := names[source.Name]; ok {
+			return apis.ErrMultipleOneOf("b.spec.sources.names")
+		}
+		// multiple sources cannot have subpath defined
+		if source.SubPath != "" {
+			if subPathExists {
+				return apis.ErrInvalidValue("b.spec.sources.subpath", source.SubPath)
+			}
+			subPathExists = true
+		}
+		names[source.Name] = ""
+
+		if source.TargetPath == "" {
+			if emptyTargetPath {
+				return apis.ErrMultipleOneOf("b.spec.sources.targetPath")
```

Comment:
can't we provide a more helpful error message here? Seems similar to subpath

Reply:
Invalid value makes more sense here. I will update